src/utils/gif.js

```diff
@@ -0,0 +1,69 @@
+/* global File */
+
+import jpegasus from 'jpegasus';
+import toArrayBuffer from 'to-arraybuffer';
+import toBufferCb from 'blob-to-buffer';
+import sharp from 'sharp';
+import JPEG from './jpeg';
+import PNG from './png';
+
+const toBuffer = blob =>
+  new Promise((resolve, reject) =>
+    toBufferCb(blob, (err, buffer) => (err ? reject(err) : resolve(buffer))),
+  );
+
+const renderGIF = async data => {
+  if (!GIF.isValid(data)) {
+    throw new TypeError(
+      'Image passed to GIF decoder appears not to be in GIF format',
+    );
+  }
+
+  if (!BROWSER) {
+    const pngBuffer = await sharp(data).toBuffer();
+    return new PNG(pngBuffer);
+  }
```

Comment:
There is an issue with the way Rollup bundles this — because it doesn’t do _dead-code elimination_ exactly, the browser builds still import the server-side `sharp` module, so — even though the `BROWSER` global is set to `true` — when it’s consuming the build Webpack processes the `import sharp from 'sharp'` statement and gives this error:

<details>
<summary>
<code>WARNING: Critical dependency: the request of a dependency is an expression</code> (expand for details)
</summary>
&nbsp;

    WARNING in ./node_modules/sharp/lib/libvips.js 51:22-69
    Critical dependency: the request of a dependency is an expression
    @ ./node_modules/sharp/lib/constructor.js
    @ ./node_modules/sharp/lib/index.js
    @ ./node_modules/@react-pdf/renderer/dist/react-pdf.browser.es.js
    @ ./src/client/components/xocs/jsx-components.pdf.js
    @ ./src/client/clientRender.pdf.js
    @ ./src/client/index.pdf.js
    @ multi (webpack)-dev-server/client?http://localhost:8082 (webpack)/hot/dev-server.js ./src/client/index.pdf ./src/client/style.scss

    WARNING in ./node_modules/sharp/lib/libvips.js 52:23-70
    Critical dependency: the request of a dependency is an expression
    @ ./node_modules/sharp/lib/constructor.js
    @ ./node_modules/sharp/lib/index.js
    @ ./node_modules/@react-pdf/renderer/dist/react-pdf.browser.es.js
    @ ./src/client/components/xocs/jsx-components.pdf.js
    @ ./src/client/clientRender.pdf.js
    @ ./src/client/index.pdf.js
    @ multi (webpack)-dev-server/client?http://localhost:8082 (webpack)/hot/dev-server.js ./src/client/index.pdf ./src/client/style.scss
</details>
<details>
<summary>
<code>ERROR: Module not found: Error: Can't resolve 'child_process'</code> (expand for details)
</summary>
&nbsp;

    ERROR in ./node_modules/detect-libc/lib/detect-libc.js
    Module not found: Error: Can't resolve 'child_process' in './node_modules/detect-libc/lib'
    @ ./node_modules/detect-libc/lib/detect-libc.js 4:16-40
    @ ./node_modules/sharp/lib/platform.js
    @ ./node_modules/sharp/lib/libvips.js
    @ ./node_modules/sharp/lib/constructor.js
    @ ./node_modules/sharp/lib/index.js
    @ ./node_modules/@react-pdf/renderer/dist/react-pdf.browser.es.js
    @ ./src/client/components/xocs/jsx-components.pdf.js
    @ ./src/client/clientRender.pdf.js
    @ ./src/client/index.pdf.js
    @ multi (webpack)-dev-server/client?http://localhost:8082 (webpack)/hot/dev-server.js ./src/client/index.pdf ./src/client/style.scss

    ERROR in ./node_modules/sharp/lib/libvips.js
    Module not found: Error: Can't resolve 'child_process' in './node_modules/sharp/lib'
    @ ./node_modules/sharp/lib/libvips.js 6:18-42
    @ ./node_modules/sharp/lib/constructor.js
    @ ./node_modules/sharp/lib/index.js
    @ ./node_modules/@react-pdf/renderer/dist/react-pdf.browser.es.js
    @ ./src/client/components/xocs/jsx-components.pdf.js
    @ ./src/client/clientRender.pdf.js
    @ ./src/client/index.pdf.js
    @ multi (webpack)-dev-server/client?http://localhost:8082 (webpack)/hot/dev-server.js ./src/client/index.pdf ./src/client/style.scss

    ERROR in ./node_modules/bindings/bindings.js
    Module not found: Error: Can't resolve 'fs' in './node_modules/bindings'
    @ ./node_modules/bindings/bindings.js 5:9-22
    @ ./node_modules/sharp/lib/constructor.js
    @ ./node_modules/sharp/lib/index.js
    @ ./node_modules/@react-pdf/renderer/dist/react-pdf.browser.es.js
    @ ./src/client/components/xocs/jsx-components.pdf.js
    @ ./src/client/clientRender.pdf.js
    @ ./src/client/index.pdf.js
    @ multi (webpack)-dev-server/client?http://localhost:8082 (webpack)/hot/dev-server.js ./src/client/index.pdf ./src/client/style.scss

    ERROR in ./node_modules/detect-libc/lib/detect-libc.js
    Module not found: Error: Can't resolve 'fs' in './node_modules/detect-libc/lib'
    @ ./node_modules/detect-libc/lib/detect-libc.js 5:18-31
    @ ./node_modules/sharp/lib/platform.js
    @ ./node_modules/sharp/lib/libvips.js
    @ ./node_modules/sharp/lib/constructor.js
    @ ./node_modules/sharp/lib/index.js
    @ ./node_modules/@react-pdf/renderer/dist/react-pdf.browser.es.js
    @ ./src/client/components/xocs/jsx-components.pdf.js
    @ ./src/client/clientRender.pdf.js
    @ ./src/client/index.pdf.js
    @ multi (webpack)-dev-server/client?http://localhost:8082 (webpack)/hot/dev-server.js ./src/client/index.pdf ./src/client/style.scss

    ERROR in ./node_modules/sharp/lib/libvips.js
    Module not found: Error: Can't resolve 'fs' in './node_modules/sharp/lib'
    @ ./node_modules/sharp/lib/libvips.js 3:11-24
    @ ./node_modules/sharp/lib/constructor.js
    @ ./node_modules/sharp/lib/index.js
    @ ./node_modules/@react-pdf/renderer/dist/react-pdf.browser.es.js
    @ ./src/client/components/xocs/jsx-components.pdf.js
    @ ./src/client/clientRender.pdf.js
    @ ./src/client/index.pdf.js
    @ multi (webpack)-dev-server/client?http://localhost:8082 (webpack)/hot/dev-server.js ./src/client/index.pdf ./src/client/style.scss

    ERROR in ./node_modules/sharp/build/Release/sharp.node 1:0
    Module parse failed: Unexpected character '�' (1:0)
    You may need an appropriate loader to handle this file type.
    (Source code omitted for this binary file)
    @ ./node_modules/sharp/lib/input.js 5:14-52
    @ ./node_modules/sharp/lib/index.js
    @ ./node_modules/@react-pdf/renderer/dist/react-pdf.browser.es.js
    @ ./src/client/components/xocs/jsx-components.pdf.js
    @ ./src/client/clientRender.pdf.js
    @ ./src/client/index.pdf.js
    @ multi (webpack)-dev-server/client?http://localhost:8082 (webpack)/hot/dev-server.js ./src/client/index.pdf ./src/client/style.scss
    ℹ ｢wdm｣: Failed to compile.
</details>

Do you know of a way to shake the non-`BROWSER` code completely out of Rollup’s browser builds? :)